bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Sleuth Kit Data Model
+ *
+ * Copyright 2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel;
+
+import java.io.Closeable;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A representation of the blackboard, a place where artifacts and their
+ * attributes are posted.
+ *
+ */
+public final class Blackboard implements Closeable {
+	private SleuthkitCase caseDb;
+
+	/**
+	 * Constructs a representation of the blackboard, a place where artifacts
+	 * and their attributes are posted.
+	 *
+	 * @param casedb The case database.
+	 */
+	public Blackboard(SleuthkitCase casedb) {
```

Comment:
This class should not have a public constructor. It should instead be a per case singleton available through a SleuthkitCase method, as in the timeline-event-mgr branch:

	public synchronized Blackboard getBlackboard() {
		if (null == blackboardInstance) {
			blackboardInstance = new Blackboard(this);
		}
		return blackboardInstance;
	}